ci/builders/mac_host_engine.json

```diff
@@ -349,31 +349,38 @@
     "archives": [
         {
             "source": "out/release/framework/FlutterMacOS.dSYM.zip",
-            "destination": "darwin-x64-release/FlutterMacOS.dSYM.zip"
+            "destination": "darwin-x64-release/FlutterMacOS.dSYM.zip",
+            "mac_codesign": false
```

Comment:
How are we planning to use this property?

Reply:
context: the files are shared for try/prod/staging are we planning to sign in the same way for all the environments?

Reply:
umm right my current understanding of global generator is that we go through the first phase of (1) producing all the global generator files, but without uploading to GCS, via `_run_global_generator`. And then the (2) upload all the files listed under the "archives" attribute to GCS. 

For global generators, my current plan it to add a step (3) of scheduling code sign luci jobs after step (2) completed, and so it would go through all the generator "archives", and select those with the "mac_codesign" attribute to schedule a subsequent code sign luci job.

Reply:
> context: the files are shared for try/prod/staging are we planning to sign in the same way for all the environments?

Right yeah my understanding is that try/prod/staging  could be set by the `environment` on the recipe side. Right I was thinking if we decide to add code sign as a step after step (2), then we would like to have a field in json to specify which archive from the "archives" might need further processing. 

My primary concern was that given my current understanding of the need to do code sign after step (2), as opposed to do it distributedly in step(1), we might lose some efficiency.

Reply:
Maybe we can simplify your approach a bit more. What do you think about this approach:

* Add a signer recipe here https://cs.opensource.google/flutter/recipes/+/main:recipes/engine_v2/
* The signer recipe receives the full json file from here: https://cs.opensource.google/flutter/engine/+/main:ci/builders/
* signer recipe use the archives section of the subbuilds and global generators to create a list of final artifact destinations.
* signer iterates over the list calling the signing utility for each of the files in the list
* signer subbuild can be triggered here https://cs.opensource.google/flutter/recipes/+/main:recipes/engine_v2/engine_v2.py;l=173 maybe using the bucket or the branch name.

Reply:
umm just a quick verification, my understanding is that a signer recipe would be used for generator files, and not including the regular sub builds produced in https://cs.opensource.google/flutter/recipes/+/main:recipes/engine_v2/engine_v2.py;l=88. which would be distributedly covered in the engine v2 / builder.py recipe. (I was thinking for these files, we would append code sign jobs at the end of builder.py)

umm for triggering generator recipe at L173, umm maybe I am wrong, I was thinking when we do subbuilds at  https://cs.opensource.google/flutter/recipes/+/main:recipes/engine_v2/engine_v2.py;l=173, maybe we are no longer doing it distributed-ly, because we trigger sub builds only after we have completed processing all the generators and archives from https://cs.opensource.google/flutter/recipes/+/main:recipes/engine_v2/engine_v2.py;l=113. Or maybe in our case it is good enough to not do it completely distributed. 

>signer recipe use the archives section of the subbuilds and global generators to create a list of final artifact destinations.

umm yeah the idea of a signer recipe sounds interesting. correct me if i am wrong, i guess the proposed approach by sir Godofredo here is to use logic in signer recipe to filter out which archive out of the "archives" should be code signed. I was thinking if we are using the "archives" section and filtering it down to a final "list", then we might still need to discern which archive need to be code signed, and in this case an attribute of `mac_codesign` in json attached to each archive might still be helpful?

Reply:
I'd recommend to keep the signing logic at the end of the orchestrator recipe to avoid having to duplicate the logic, this should not impact a lot since the files will need to be uploaded and downloaded in the signer machine.

I would say the aproach can be even simpler if we assume everything coming out of the mac builders need to be codesigned and then filtering based on the codesigning metadata. Downloading all the builds have a cost <1m, unzipping may be costly but we can optimize later by just reading the zip headers to see if it contains metadata files.

Reply:
Thanks for the idea! I made an attempt based on this idea in https://flutter-review.googlesource.com/c/recipes/+/34900. I posted two questions I had on the recipe side that I could not quite figure out. Besides those questions, in recipe I used a list of keywords ["dSYM", "objcdoc", "nobitcode"] to circumvent the signing property in json. I shifted the logic from under engine v2 folder to within the main v2 recipe, because I think a recipe can only trigger a recipe module and seems like it cannot trigger another recipe.